tests/currency_network/test_gas_costs_currency_network.py

```diff
@@ -99,205 +66,157 @@ def currency_network_contract_with_trustlines(web3, accounts):
     return contract
 
 
-def find_gas_values_for_call(web3, contract_call, transaction_options=None):
-    if transaction_options is None:
-        transaction_options = {}
-
-    gas_limit = 20_000
-    tx_success = False
-
-    while not tx_success:
-        gas_limit += 1_000
-        transaction_options["gas"] = gas_limit
-        try:
-            tx_hash = contract_call.transact(transaction_options)
-            tx_success = web3.eth.waitForTransactionReceipt(tx_hash, 5).status == 1
-        except ValidationError as e:
-            if "Insufficient gas" not in e.args:
-                raise e
-            tx_success = False
-
-    gas_cost = get_gas_costs(web3, tx_hash)
-    return GasValues(limit=gas_limit, cost=gas_cost)
-
-
-def assert_transaction_successful(web3, tx_hash):
-    tx_receipt = web3.eth.waitForTransactionReceipt(tx_hash, 5)
-    assert tx_receipt.status == 1, "Transaction failed, check gas limit"
-
-
 def test_cost_transfer_0_mediators(
-    web3, currency_network_contract_with_trustlines, accounts, table
+    web3, currency_network_contract_with_trustlines, accounts, gas_values_snapshot
 ):
     contract = currency_network_contract_with_trustlines
     A, B, *rest = accounts
     # Verify that we test the most gas expensive case where we change the balance from 0 to non-zero
     assert contract.functions.balance(A, B).call() == 0
-    tx_hash = contract.functions.transfer(100, 2, [A, B], EXTRA_DATA).transact(
-        {"from": A, "gas": GAS_LIMIT_TRANSFER_0_MEDIATOR}
-    )
-    assert_transaction_successful(web3, tx_hash)
-    gas_cost = get_gas_costs(web3, tx_hash)
-    report_gas_costs(
-        table, "0 hop transfer", gas_cost, limit=GAS_COST_TRANSFER_0_MEDIATOR
+    gas_values_snapshot.assert_gas_values_for_call(
```

Comment:
This is called `assert_gas_values_for_call` but it also store them if they are new, or updates them if you used `--update` I'm not sure that is clean.

Reply:
same as above, this is the same in jest, where it is called expect(x).toMatchSnapshot();
So I could rename it to `assert_gas_value_snapshot_for_call`? is that better?
Or `assert_gas_values_match_for_call`

Reply:
Yeah, I like `assert_gas_values_match_for_call`, but you're right that I only find it weird because I didn't know about snapshot testing and had to figure out the intent of the code from the code itself.